### PR TITLE
Templated components and Razor Templates

### LINKED ISSUE #251: Templated components and Razor Templates
Add content on templated components:
- `RenderFragment<T>`, Razor templates
- Generic components
- Type inference
- Context parameter

https://github.com/aspnet/Blazor/pull/1360
https://github.com/aspnet/Blazor/pull/1404
https://github.com/aspnet/Blazor/pull/1417
https://github.com/aspnet/Blazor/pull/1453
https://github.com/aspnet/Blazor/pull/1470

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #251 

* Not sure if I butchered your *GenericComponent* example by making it into an implemented template (*ListViewTemplate*). If I need to back off on the "template" language because generics are more widely applicable, then let me know what you advise: dial it back to just a component or say in the text that generics can be used with any component (not only templates). Depending on how this plays out, we'll adjust the heading for the generics content from H3 to H2 as well.
* Takes the common sample to 0.6.0.
* I only breezed thru the issues/PRs and relied heavily on your blog post synthesis (i.e., *I stole ur text!*), so there might be some bits missing here.
* ... and this doesn't address the *Use server-side Blazor with the Azure SignalR Service* section of your post.

### CODE REVIEW

File: docs/common/samples/2.x/BlazorSample/Pages/TemplatedComponents.cshtml [hunk 1]
```diff
@@ -0,0 +1,111 @@
+@page "/TemplatedComponents"
+
+<h1>Templated Components</h1>
+
+<p>Templated components are components that accept one or more UI templates as parameters, which can then be used as part of the component's rendering logic. Templated components allow you to author higher-level components that are more reusable than regular components.</p>
+
+<p>The following table template examples use the template defined in <em>Templates/TableTemplate.cshtml</em>.</p>
```
Comment (danroth27) [inline docs/common/samples/2.x/BlazorSample/Pages/TemplatedComponents.cshtml]: use the component defined in `<em>Components/TableTemplate.cshtml</em>`

File: docs/common/samples/2.x/BlazorSample/Pages/TemplatedComponents.cshtml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+<h2>Implicit parameters</h2>
```
Comment (danroth27) [inline docs/common/samples/2.x/BlazorSample/Pages/TemplatedComponents.cshtml]: Implicit template parameters

File: docs/common/samples/2.x/BlazorSample/Pages/TemplatedComponents.cshtml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    </TableHeader>
+    <RowTemplate>
+        <td>@context.PetId</td>
+        <td>@context.Name</td>
+    </RowTemplate>
+</TableTemplate>
+
+<h2>Explicit parameters</h2>
+
+<p>You can change the parameter name using the <code>Context</code> attribute on the child element.</p>
```
Comment (danroth27) [inline docs/common/samples/2.x/BlazorSample/Pages/TemplatedComponents.cshtml]: the template parameter name

File: docs/components/index.md [hunk 4]
```diff
@@ -458,3 +458,132 @@ The following example shows using the `MarkupString` type to add a block of stat
     string myMarkup = "<p class='markup'>This is a <em>markup string</em>.</p>";
 }
 ```
+
+## Templated components
+
+*Templated components are available in Blazor 0.6.0 or later.*
```
Comment (danroth27) [inline docs/components/index.md]: I don't think we need to version our content at this point. Things are moving too fast for that. I would remove this.

File: docs/components/index.md [hunk 5]
```diff
@@ -458,3 +458,132 @@ The following example shows using the `MarkupString` type to add a block of stat
...
+<ListViewTemplate Items="@pets" TItem="Pet">
+    <ItemTemplate Context="pet">
+        <li>@pet.Name</li>
+    </ItemTemplate>
+</ListViewTemplate>
+```
+
+## Razor templates
+
+*Razor templates are available in Blazor 0.6.0 or later.*
```
Comment (danroth27) [inline docs/components/index.md]: Remove

**Comment by guardrex** [comment]
@danroth27 This should be ready to merge. Is it a **_'GO' for throttle UP?!_** 🚀

**Pull request merged**